borg/keymanager.py

```diff
@@ -0,0 +1,211 @@
+        with open(args.path, 'r') as fd:
+            file_first_line = fd.read(len(first_line))
+            if file_first_line != first_line:
+                if file_first_line.startswith(file_id):
+                    raise NotABorgKeyFile()
+                else:
+                    raise RepoIdMismatch()
+            self.keyblob = fd.read()
+
+        self.store_keyblob(args)
+
+    def import_paperkey(self, args):
+        # imported here because it has global side effects
+        import readline
+
+        try:
+            repoid = hexlify(self.repository.id).decode('ascii')[:18]
+
+            while True:
+                idline = input('id: ')
+                try:
+                    (data, checksum) = idline.replace(' ', '').split('-')
+                except ValueError:
+                    print('each line must contain exactly one \'-\', try again')
+                    continue
+
+                try:
+                    (id_lines, id_repoid, id_complete_checksum) = data.split('/')
+                except ValueError:
+                    print('the id line contains three \'/\', try again')
+
+                if sha256_truncated(data.lower().encode('ascii'), 2) != checksum:
+                    print('line checksum did not match, try same line again')
+                    continue
+
+                try:
+                    lines = int(id_lines)
+                except ValueError:
+                    print('internal error while parsing length')
+
+                break
+
+            if repoid != id_repoid:
+                raise RepoIdMismatch()
+
+            result = b''
+
+            idx = 1
+            while True:
+                inline = input('{0:2d}: '.format(idx))
+                inline = inline.replace(' ', '')
+                try:
+                    (data, checksum) = inline.split('-')
+                except ValueError:
+                    print('each line must contain exactly one \'-\', try again')
+                    continue
+
+                try:
+                    part = unhexlify(data)
+                except binascii.Error:
+                    print('only characters 0-9 and a-f and \'-\' are valid, try again')
+                    continue
+
+                if sha256_truncated(idx.to_bytes(2, byteorder='big') + part, 2) != checksum:
+                    print('line checksum did not match, try line {0} again'.format(idx))
+                    continue
+
+                result += part
+                if idx == lines:
+                    break
+                idx += 1
+
+        except EOFError:
+            print('\n - aborted')
+            return
+
+        if sha256_truncated(result, 12) != id_complete_checksum:
+            print('The overall checksum did not match, aborting.')
+            return
+
+        self.keyblob = '\n'.join(textwrap.wrap(b2a_base64(result).decode('ascii'))) + '\n'
+
+        self.store_keyblob(args)
```

Comment:
yes, it seems that code didn‘t get the mandatory reduce whitespace pass needed for borg^^. Should be better now.